Fixes mbedTLS support to link properly and not include libssl.so

Tracked in upstream PR https://github.com/libgit2/libgit2/pull/4678

NOTE: libgit2 has switched its CI to Azure Pipelines. The aforementioned PR makes modifications
to the Travis YAML file, which has since been removed, causing patch conflicts. That part of
the diff has thus been removed here.

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 2b82bb325..2deed5f87 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -284,8 +284,13 @@ ELSEIF(SHA1_BACKEND STREQUAL "CommonCrypto")
 ELSEIF (SHA1_BACKEND STREQUAL "mbedTLS")
 	ADD_FEATURE_INFO(SHA ON "using mbedTLS")
 	SET(GIT_SHA1_MBEDTLS 1)
-	FILE(GLOB SRC_SHA1 src/hash/hash_mbedtls.c)
-	LIST(APPEND LIBGIT2_PC_REQUIRES "mbedtls")
+	FILE(GLOB SRC_SHA1 hash/hash_mbedtls.c)
+	LIST(APPEND LIBGIT2_INCLUDES ${MBEDTLS_INCLUDE_DIR})
+	LIST(APPEND LIBGIT2_LIBS ${MBEDTLS_LIBRARIES})
+	# mbedTLS has no pkgconfig file, hence we can't require it
+	# https://github.com/ARMmbed/mbedtls/issues/228
+	# For now, pass its link flags as our own
+	LIST(APPEND LIBGIT2_PC_LIBS ${MBEDTLS_LIBRARIES})
 ELSE()
 	MESSAGE(FATAL_ERROR "Asked for unknown SHA1 backend ${SHA1_BACKEND}")
 ENDIF()
